mlflow/gateway/handlers.py

```diff
@@ -0,0 +1,216 @@
+class Model(BaseModel):
+    name: Optional[str] = None
+    provider: Union[str, Provider] = Provider.UNSPECIFIED_PROVIDER
+    config: Optional[Dict[str, Any]] = None
+
+    @validator("provider", pre=True)
+    def validate_provider(cls, value):
+        if isinstance(value, Provider):
+            return value
+        return (
+            Provider[value.upper()]
+            if value.upper() in Provider.__members__
+            else Provider.UNSPECIFIED_PROVIDER
+        )
+
+    @validator("config", pre=True)
+    def validate_config(cls, config, values):
+        provider = values.get("provider")
+        if provider:
+            config_type = config_types[provider]
+            config_instance = config_type(**config)
+
+            # validate that the configuration fields are defined properly
+            _config_instance_validation(config_instance, provider)
+
+            return config_instance
+        else:
+            raise MlflowException(
+                "A provider must be provided for each gateway route.",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+
+# pylint: disable=no-self-argument
+class RouteConfig(BaseModel):
+    name: str
+    type: RouteType = RouteType.UNSPECIFIED
+    model: Model
+
+    @validator("name")
+    def validate_endpoint_name(cls, route_name):
+        if not is_valid_endpoint_name(route_name):
+            raise MlflowException(
+                "The route name provided contains disallowed characters for a url endpoint. "
+                f"'{route_name}' is invalid. Names cannot contain spaces or any non "
+                "alphanumeric characters other than hyphen and underscore",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        return route_name
+
+    @validator("model", pre=True)
+    def validate_model(cls, model):
+        if model:
+            model_instance = Model(**model)
+            if model_instance.provider in PROVIDERS and model_instance.config is None:
+                raise MlflowException(
+                    "A config must be supplied when setting a provider. The provider entry for "
+                    f"{model_instance.provider} is incorrect.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+        return model
+
+    @validator("type", pre=True)
+    def validate_route_type(cls, value):
+        if value in RouteType._value2member_map_:
+            return value
+        return RouteType.UNSPECIFIED.value
+
+
+class Route(BaseModel):
+    name: str
+    type: RouteType
+    model: ModelInfo
+
+
+def _load_gateway_config(path: Union[str, Path]) -> List[RouteConfig]:
+    """
+    Reads the gateway configuration yaml file from the storage location and returns an instance
+    of the configuration _GatewayConfig class
+    """
+    if isinstance(path, str):
+        path = Path(path)
+    configuration = yaml.safe_load(path.read_text())
+    return parse_obj_as(List[RouteConfig], configuration)
```

Comment:
We also need to verify that there aren't any duplicate route names. Are we doing that?

Reply:
adding checks!